app/models/TraderProfile.scala

```diff
@@ -53,12 +53,12 @@ object TraderProfile {
       Right(traderProfile.niphlNumber)
     ).parMapN(TraderProfile.apply)
 
-  private def getOptionallyRemovedPage[A](
+  def getOptionallyRemovedPage(
```

Comment:
so i'm going to stick a spanner in the works and say that the same method should be used for the niphl stuff too... ARGH

Reply:
that way you don't have the defaults for nirms (which feel wrong)

Reply:
Marcy found an issue with using it on Niphls, so either the generic implementation will change slightly or Nphls will use something else